django/db/models/base.py

```diff
@@ -660,6 +660,25 @@ def _set_pk_val(self, value):
                 setattr(self, parent_link.target_field.attname, value)
         return setattr(self, self._meta.pk.attname, value)
 
+    def _validate_force_insert(self, force_insert):
+        if isinstance(force_insert, bool) or not force_insert:
+            return force_insert
+
+        if not isinstance(force_insert, tuple):
+            raise ValidationError("force_insert must be a bool or tuple")
+
+        if any(not issubclass(item, Model) for item in force_insert):
+            raise ValidationError(
+                "All the items passed in force_insert must be a "
+                "subclass of models.Model"
+            )
+
+        if any(model == Model for model in force_insert):
+            model_tuple = (Model,)
+            return model_tuple
```

Comment:
Is this a form of optimization to retrieve the _oldest_ common base?

Reply:
Yes, It is, as it would reduce the number of checks later on in every recursive calls.
```
force_insert_parent = force_insert and any(
     issubclass(parent, model_class) for model_class in force_insert
)
```